src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1217,6 +1217,11 @@ def set_up_existing_container_update(self):
             # Fails if all containers are removed
             while (len(self.new_containerapp["properties"]["template"]["containers"]) > 1):
                 self.new_containerapp["properties"]["template"]["containers"].pop()
+            if not self.get_argument_name():
+                raise RequiredArgumentMissingError("The ContainerApp name is required to update it.")
+            # Set the container name to the container app name if force_single_container_updates is set
+            self.set_argument_container_name(self.get_argument_name())
+            safe_set(self.new_containerapp, "properties", "template", "containers", 0, "name", value=self.get_argument_container_name())
```

Comment:
Just want to be 100% sure that we are guaranteed here to have at least 1 container, and so we know this index "0" will exist?

Reply:
Yes,  the container app is guaranteed to have at least one container. But added validation here just in case.
@Greedygre  I can remove if this is a redundant check.

Reply:
@snehapar9 
In the previous logic in `containerapp_up_logic`
The input argument contains `container_name=name`, why you need to set the container name Explicitly?